shared/AppInsightsCommon/src/StorageHelperFuncs.ts

```diff
@@ -5,22 +5,23 @@ import {
     IDiagnosticLogger, _eInternalMessageId, _throwInternal, dumpObj, eLoggingSeverity, getExceptionName, getGlobal, getGlobalInst,
     isNullOrUndefined, objForEachKey
 } from "@microsoft/applicationinsights-core-js";
+import { ICachedValue, createCachedValue, objGetOwnPropertyDescriptor } from "@nevware21/ts-utils";
 import { StorageType } from "./Enums";
 
-let _canUseLocalStorage: boolean = undefined;
-let _canUseSessionStorage: boolean = undefined;
+let _canUseLocalStorage: boolean = null;
+let _canUseSessionStorage: boolean = null;
 let _storagePrefix: string = "";
 
+// Create cached values for verified storage objects to avoid repeated checks
+let _verifiedLocalStorage: ICachedValue<Storage> = createCachedValue<Storage>(null);
+let _verifiedSessionStorage: ICachedValue<Storage> = createCachedValue<Storage>(null);
+
 /**
  * Gets the localStorage object if available
  * @returns {Storage} - Returns the storage object if available else returns null
  */
 function _getLocalStorageObject(): Storage {
-    if (utlCanUseLocalStorage()) {
-        return _getVerifiedStorageObject(StorageType.LocalStorage);
-    }
-
-    return null;
+    return _verifiedLocalStorage.value(() => _getVerifiedStorageObject(StorageType.LocalStorage));
```

Comment:
Review the interface there is no "value" function.
This should be
if (!_verifiedLocalStorate) {
    _verifiedLocalStorage = createCachedValue(_getVerifiedStorageObject(StorageType.LocalStorage));
}
return _verifiedLocalStorage.v;